src/components/Planning/GroupForm/GroupProjectSearch.tsx

```diff
@@ -0,0 +1,155 @@
+import { Paragraph } from '@/components/shared';
+import { FreeSearchFormItem, IFreeSearchResult, IListItem, IOption } from '@/interfaces/common';
+import { getProjectsWithFreeSearch, getProjectsWithParams } from '@/services/projectServices';
+import { arrayHasValue, listItemToOption } from '@/utils/common';
+import { Tag } from 'hds-react/components/Tag';
+import { SearchInput } from 'hds-react/components/SearchInput';
+import _ from 'lodash';
+import { FC, memo, useCallback, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { UseFormGetValues } from 'react-hook-form';
+import { IGroupForm } from '@/interfaces/formInterfaces';
+// Build a search parameter with all the choices from the search form
+
+const buildQueryParamString = (state: IQueryParamsState, projectName: string): string => {
+  const searchParams = [];
+  for (const [key, value] of Object.entries(state)) {
+    switch (key) {
+      case 'masterClass':
+      case 'class':
+      case 'subClass':
+      case 'district':
+      case 'division':
+      case 'subDivision':
+        value && searchParams.push(`${key}=${value}`);
```

Comment:
Could this be done without a switch case, just a simple loop?

Reply:
Changed